src/primitives.coffee

```diff
@@ -85,6 +85,19 @@ class Collection extends ElementType
 
   add: (val) -> @push val
 
+  findElements: (cond) ->
+    elements = @content.reduce (results, el) ->
+      return results.concat([el.findElements(cond)]) if el.elementType() in ['array', 'object']
+      return results.concat([el]) if cond(el)
+      return results
+    , []
+    _.flatten elements, true
+
+  find: (cond) ->
```

Comment:
Is this really how find should work? That is finds stuff with no context of where it was in the structure? Wondering the use of this? Further, seems recursion could solve this as well.